src/Payments.Core/Extensions/StringExtensions.cs

```diff
@@ -29,5 +30,24 @@ public static string SafeTruncate(this string value, int max)
 
             return value.Substring(0, max);
         }
+
+        public static string SafeRegexRemove(this string value, string regEx = @"[^0-9a-zA-Z\.\-\' ]+")
+        {
+            if (string.IsNullOrWhiteSpace(value))
+            {
+                return value;
+            }
+            
+            try
+            {
+                return Regex.Replace(value, regEx, string.Empty);
+            }
+            catch (Exception e)

```

Comment:
just FYI if you aren't going to use the exception i think you can just leave out the variable